File: caracal-node/src/adapters/websocket.ts
/**
 * Copyright (C) 2026 Garudex Labs. All Rights Reserved.
 * Caracal, a product of Garudex Labs
 *
 * WebSocket transport adapter (placeholder — coming in v0.4).
 */

import { BaseAdapter, SDKRequest, SDKResponse } from './base';

export class WebSocketAdapter extends BaseAdapter {
  constructor(private readonly options: { url: string }) {
    super();
  }

  async send(_request: SDKRequest): Promise<SDKResponse> {
    throw new Error('WebSocket adapter coming in v0.4');
  }

  close(): void {
    // no-op
  }

  get isConnected(): boolean {
    return false;
  }
}

File: caracal/sdk/adapters/websocket.py
"""
Copyright (C) 2026 Garudex Labs.  All Rights Reserved.
Caracal, a product of Garudex Labs

WebSocket transport adapter (placeholder — coming in v0.4).
"""

from __future__ import annotations

from caracal.sdk.adapters.base import BaseAdapter, SDKRequest, SDKResponse


class WebSocketAdapter(BaseAdapter):
    """Real-time WebSocket transport (not yet implemented).

    Args:
        url: WebSocket endpoint (e.g. ``wss://caracal.internal:8443``).
    """

    def __init__(self, url: str) -> None:
        self._url = url

    async def send(self, request: SDKRequest) -> SDKResponse:
        raise NotImplementedError("WebSocket adapter coming in v0.4")

    def close(self) -> None:
        pass

    @property
    def is_connected(self) -> bool:
        return False
